src/Http/Controllers/Inertia/Concerns/ConfirmsTwoFactorAuthentication.php

```diff
@@ -77,7 +77,8 @@ protected function hasJustBegunConfirmingTwoFactorAuthentication(Request $reques
      */
     protected function neverFinishedConfirmingTwoFactorAuthentication(Request $request, $currentTime)
     {
-        return is_null($request->user()->two_factor_confirmed_at) &&
+        return ! array_key_exists('code', $request->session()->getOldInput()) &&
+            is_null($request->user()->two_factor_confirmed_at) &&
             $request->session()->get('two_factor_confirming_at', 0) != $currentTime;
```

Comment:
I'm wondering if this like is the culprit and we need to allow a window of time here instead of checking with `!=`?

Reply:
This is correct @driesvints

Reply:
You are right. That works as well. For testing I changed the code to the following:

```
        return is_null($request->user()->two_factor_confirmed_at) &&
            $currentTime - $request->session()->get('two_factor_confirming_at', 0) > 60 * 5;
```

Is that the general direction you had in mind?

Do you prefer the window of time to be configurable (e.g. `config/jetstream.php` as it is not really part of `fortify.php` or `auth.php`) instead of being hard-coded?

If we choose a time window that is too short, we will face problems sooner or later. I see it coming that users search for their mobile phone or get distracted. Then they scan or scanned a meanwhile outdated QR code and get frustrated. I might as well focus too much on that unimportant detail here ;)

If we choose to make the window long enough this might not be a concern.

Reply:
@ManuelLeiner I think we can copy the timing window from livewire which uses this:
`$maximumSecondsSinceConfirmation = $maximumSecondsSinceConfirmation ?: config('auth.password_timeout', 900);`

Reply:
I will have a look at it. Thanks for the input. :)

Reply:
@taylorotwell are you sure? I thought we implemented that time session to allow some time for the user to attempt setting up the 2FA. Checking that the saved time session matches exactly with the current time seems to defeat that purpose? Maybe I'm not seeing the full picture here.

Reply:
I was trying tos ay I don't think this code should be changed.